pkg/apis/workload/v1alpha1/types.go

```diff
@@ -56,8 +56,9 @@ type KubernetesApplicationSpec struct {
 	ResourceSelector *metav1.LabelSelector `json:"resourceSelector"`
 
 	// ClusterSelector selects the clusters to which this application may be
-	// scheduled.
-	ClusterSelector *metav1.LabelSelector `json:"clusterSelector,omitempty"`
+	// scheduled. Leave both match labels and expressions empty to match any
+	// cluster.
+	ClusterSelector *metav1.LabelSelector `json:"clusterSelector"`
```

Comment:
I suspect it might be a more intuitive user experience to match any cluster by omitting `clusterSelector` completely, but this would require us to work around `metav1.LabelSelectorAsSelector()`'s logic.